apps/ngrx/effect-selector/src/app/app.component.ts

```diff
@@ -1,72 +1,29 @@
-import { AsyncPipe, NgFor } from '@angular/common';
 import {
   ChangeDetectionStrategy,
   Component,
   OnInit,
   inject,
 } from '@angular/core';
 import { Store } from '@ngrx/store';
-import { loadActivities } from './store/activity/activity.actions';
-import { ActivityType } from './store/activity/activity.model';
-import { selectActivities } from './store/activity/activity.selectors';
-import { loadStatuses } from './store/status/status.actions';
-import { selectAllTeachersByActivityType } from './store/status/status.selectors';
-import { loadUsers } from './store/user/user.actions';
+import { AppActions } from './app.actions';
+import { AppStore } from './app.config';
+import { ActivitySelectors } from './store/activity/activity.selectors';
 
 @Component({
   selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrl: './app.component.scss',
   standalone: true,
-  imports: [NgFor, AsyncPipe],
-  template: `
-    <h1>Activity Board</h1>
-    <section>
-      <div class="card" *ngFor="let activity of activities$ | async">
-        <h2>Activity Name: {{ activity.name }}</h2>
-        <p>Main teacher: {{ activity.teacher.name }}</p>
-        <span>All teachers available for : {{ activity.type }} are</span>
-        <ul>
-          <li
-            *ngFor="
-              let teacher of getAllTeachersForActivityType$(activity.type)
-                | async
-            ">
-            {{ teacher.name }}
-          </li>
-        </ul>
-      </div>
-    </section>
-  `,
-  styles: [
-    `
-      section {
-        display: grid;
-        grid-template-columns: repeat(3, minmax(0, 1fr));
-        gap: 2px;
-      }
-
-      .card {
-        display: flex;
-        flex-direction: column;
-        border: solid;
-        border-width: 1px;
-        border-color: black;
-        padding: 2px;
-      }
-    `,
-  ],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class AppComponent implements OnInit {
-  private store = inject(Store);
+  readonly #store: Store<AppStore> = inject(Store<AppStore>);
```

Comment:
you don't have to type your store

Reply:
why not?

Reply:
it's always better to let typescript infer the type.